### DISCUSSION TITLE
Support an easy way to access fields by adding collection-field to DOM classes

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#19372] [Feature Requests]
### Summary

Since Directus includes collection and field for inputs, it would be useful to get it also for relational fields.
This would help for Custom CSS as well for writing automatic tests.
Without this information, we have to rely on `nth-child` which may vary when changing the order of fields or when introducing new fields.

### Basic Example

We have:
<input data-v-f5e6fd87="" loading="false" batch-mode="false" batch-active="false" width="0" **collection="test2"** **field="input3"** field-data="[object Object]" primary-key="1" autocomplete="off" type="text" step="1">

Wanted:
< div data-v-81599a50="" data-v-03528dd7="" class="many-to-one" loading="false" batch-mode="false" batch-active="false" _**collection="the_related_collection_name4"** **field="the_related_field_name4"**_ width="full" type="uuid" field-data="[object Object]" primary-key="1">

< div data-v-d39e9e3e="" data-v-03528dd7="" class="many-to-many" loading="false" batch-mode="false" batch-active="false" _**collection="the_related_collection_name5"** **field="the_related_field_name5"**_ type="alias" field-data="[object Object]">

### Motivation

Be able to easily address custom CSS
```
div[collection="the_related_collection_name4"][field="the_related_field_name4"] .input {
  ...
}
```
which is a stable even if we change the order of the fields or insert a new field before.

and in e2e test
instead of
`cy.get(':nth-child(5) > .interface > .v-menu > .v-menu-activator > .v-input > .input > .append > .v-icon > i').click()`
have
`cy.get('div[collection="the_related_collection_name4"][field="the_related_field_name4"] .input > .append > .v-icon > i').click()`

I currently have a hundred of nth- in my test. This is hard to maintain.

### Detailed Design

<div data-v-81599a50="" data-v-03528dd7="" class="many-to-one" loading="false" batch-mode="false" batch-active="false" _**collection="the_related_collection_name4"** **field="the_related_field_name4"**_ width="full" type="uuid" field-data="[object Object]" primary-key="1">

<div data-v-d39e9e3e="" data-v-03528dd7="" class="many-to-many" loading="false" batch-mode="false" batch-active="false" _**collection="the_related_collection_name5"** **field="the_related_field_name5"**_ type="alias" field-data="[object Object]">

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

None

### Alternatives

Build an extension for every relational type just to include _**collection**_ and _**field**_ in the template: that's heavy.

### Adoption Strategy

No UI changes. Immediate adoption.

**Comment by rijkvanzanten** [comment]
For what it's worth.. collection and field being set as attributes is actually a bug, not intended behavior (as it's invalid HTML). It's because of Vue's default behavior of passing any unspecified props as attributes onto the root element of the component. In this case, it "works" for some fields as those aren't using the collection and field props.

**Reply by Pachat** [reply]
In this case, how to obtain in the DOM a means of identifying the elements?

Would adding `collection-field` to the class be a solution?

That would help to easily address custom CSS
```
.collection-field .input {
    ...
}
```
stable even if we change the order of the fields or insert a new field before.

and in the e2e test
instead of
`cy.get(':nth-child(5) > .interface > .v-menu > .v-menu-activator > .v-input > .input > .append > .v-icon > i').click( )`
have
`cy.get('.collection-field .input > .append > .v-icon > i').click()`

**Reply by rijkvanzanten** [reply]
Lets treat this feature request as that 👍🏻

**Reply by Pachat** [reply]
Following your comment, I changed the OP title to something more appropriate.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
